### PR TITLE
Support overrides

### PR DESCRIPTION
**Pull request by vixalien**
also move gobject.object overrides to new file
allowing to construct objects by extending the existing object

### CODE REVIEW

**Comment by vixalien** [comment]
This pattern is similar to what GJS does already, for example: https://gitlab.gnome.org/GNOME/gjs/-/blob/HEAD/modules/core/overrides/GLib.js

**Comment by vixalien** [comment]
This PR also needs to get merged later, converting to draft.

**Comment by vixalien** [comment]
@ahgilak, I believe this is ready for review. I would like to know what you feel about this. This will be the basis of other very cool stuff I'm working on, such as #23 and #20

File: src/types/object.js [hunk 1]
```diff
@@ -2,22 +2,19 @@ import g from "../bindings/mod.js";
...
   const parent = g.object_info.get_parent(info);
 
   if (parent) {
     const gType = g.registered_type_info.get_g_type(parent);
 
     const ParentClass = objectByGType(gType);
-    Object.setPrototypeOf(target.prototype, ParentClass.prototype);
-    //Object.assign(target.__signals__, ParentClass.__signals__);
 
-    g.base_info.unref(parent);
```
Comment (ahgilak) [inline src/types/object.js:20]: is there a reason to keep the parent reference?

**Comment by ahgilak** [comment]
Are you sure this is ready?
Everything seems broken.

File: src/types/object.js (hunk 1 above)
Comment (vixalien) [reply src/types/object.js:20]: Yes, afaict if the reference is dropped, it won't be possible to construct the child types. Although, I'm unsure about this and some testing may be required. I can test it after #25 is reviewed, as that depends on it.

**Comment by vixalien** [comment]
Wait. I see what happened now. I got errors while trying to execute the files in the `examples/` folder, and it turns out they were using an older version of `deno_gi` because they got it from https://gir.deno.dev. Could you add versioning to the domain like it's done for `https://deno.land/x`? It may even be a good idea to make the site's code open source if possible.

This PR is ready for re-review, as it works, please let me know what you think.

**Comment by ahgilak** [comment]
> Hmm. I see the examples are not working. However, they are not working in the `main` branch either

Only `stylesheet.ts` is broken on main, other examples are fine.

**Comment by ahgilak** [comment]
main purpose of `gir.deno.dev` is to add typescript definitions, besides that it only loads the library and nothing else.

you can test your local changes by adding the following deno.json file:
```json
{
  "imports": {
    "https://github.com/ahgilak/deno_gi/raw/main/mod.ts": "./mod.ts"
  }
}
```

**Comment by ahgilak** [comment]
This is the error comes up while running the examples in current PR
```
error: Uncaught (in promise) TypeError: app.on is not a function
app.on("activate", activate);
    ^
```

**Comment by vixalien** [comment]
> No I meant other examples are working fine on main branch, but broken in this PR.

Yes, I know :)

I was saying that that problem is indeed exposed by this PR, but the fix is in a separate PR because it handles other cases too of this bug. I will include the fix in this PR though, to streamline things.

**Comment by vixalien** [comment]
fixed everything. please take a look!

File: src/types/object.js (hunk 1 above)
Comment (ahgilak) [reply src/types/object.js:20]: Did you test this?

Comment (vixalien) [reply src/types/object.js:20]: I just tested this, and it seems it's indeed uneeded.

**Comment by vixalien** [comment]
Thanks!

**Pull request merged**